s3stream/src/main/java/com/automq/stream/s3/S3StreamClient.java

```diff
@@ -366,8 +369,15 @@ private void compactV1(CompactionHint hint, long now) {
         }
 
         public void compact(StreamObjectCompactor.CompactionType compactionType) {
-            StreamObjectCompactor task = StreamObjectCompactor.builder().objectManager(objectManager).stream(this)
-                .objectStorage(objectStorage).maxStreamObjectSize(config.streamObjectCompactionMaxSizeBytes()).build();
+            StreamObjectCompactor task = StreamObjectCompactor.builder()
+                .objectManager(objectManager)
+                .stream(this)
+                .objectStorage(objectStorage)
+                .maxStreamObjectSize(config.streamObjectCompactionMaxSizeBytes())
+                .minorV1CompactionThreshold(MINOR_V1_COMPACTION_SIZE)
+                .majorV1CompactionSkipSmallObject(MAJOR_V1_COMPACTION_SKIP_SMALL_OBJECT)
```

Comment:
We can skip the small object logic for objectsCount < MAJOR_V1_COMPACTION_MAX_OBJECT_THRESHOLD instead of by a switch.